ios/browser/api/history/brave_browsing_history_driver.h

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_HISTORY_BRAVE_BROWSING_HISTORY_DRIVER_H_
+#define BRAVE_IOS_BROWSER_API_HISTORY_BRAVE_BROWSING_HISTORY_DRIVER_H_
+
+#import <Foundation/Foundation.h>
+#include <vector>
+
+#include "base/macros.h"
+#include "base/memory/weak_ptr.h"
+
+#include "components/history/core/browser/browsing_history_driver.h"
+#include "components/history/core/browser/browsing_history_service.h"
+
+#include "url/gurl.h"
+
+class ChromeBrowserState;
+
+namespace history {
+class HistoryService;
+}
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol BraveHistoryDriverDelegate
+
+/// Tells the consumer that the result of a history query has been retrieved
+/// @param results - Results of the query (Sorted)
+/// @param queryResultsInfo - Detailed Query Results
+/// @param continuationClosure - Pagination lambda
+/// If called, will continue fetching results where last left off.
+- (void)historyQueryWasCompletedWithResults:
+    (const std::vector<history::BrowsingHistoryService::HistoryEntry>&)results
+                   queryResultsInfo:(const history::BrowsingHistoryService::
+                                         QueryResultsInfo&)queryResultsInfo
+                continuationClosure:(base::OnceClosure)continuationClosure;
+
+/// Tells the consumer that history entries
+/// have been deleted from a different client
+- (void)historyWasDeleted;
+
+@end
+
+class BraveBrowsingHistoryDriver : public history::BrowsingHistoryDriver {
```

Comment:
Rename to `BrowserHistoryDriverIOS` to match Chromium naming conventions for iOS counterparts (along with file name: `browsing_history_driver_ios.h/mm`

Reply:
Renamed.